File: android/app/src/main/java/com/androidturbomodules/newarchitecture/modules/Bike.java
package com.androidturbomodules.newarchitecture.modules;

import androidx.annotation.NonNull;

import com.androidturbomodules.codegen.NativeBikeSpec;
import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.ReactApplicationContext;

public class Bike extends NativeBikeSpec {

    public static final String NAME = "Bike";

    public Bike(ReactApplicationContext reactContext) {
        super(reactContext);
    }

    @Override
    public void calculateAverage(double average, Promise promise) {
        promise.resolve(average*10);
    }

    @Override
    public String getBikeName(String bikeName) {
        return "Bike Name is "+bikeName;
    }

    @NonNull
    @Override
    public String getName() {
        return NAME;
    }
}

File: android/app/src/main/java/com/androidturbomodules/newarchitecture/modules/Bike.kt
package com.androidturbomodules.newarchitecture.modules

import com.androidturbomodules.codegen.NativeBikeSpec
import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext

class Bike(reactContext: ReactApplicationContext?) : NativeBikeSpec(reactContext) {

    companion object {
        const val NAME = "Bike"
    }

    override fun getName(): String {
        return NAME
    }

    override fun calculateAverage(average: Double, promise: Promise?) {
        promise?.resolve(average*10)
    }

    override fun getBikeName(bikeName: String?): String {
        return "Bike Name is $bikeName"
    }
}
